### PR TITLE
Proposal: Introduce zeitwerk autoloading

### PR DESCRIPTION
**Pull request by ilvez**
Instead of manual autoloading use [Zeitwerk](https://github.com/fxn/zeitwerk). It forces upon conventional layout, but instead gives zero hassle and adjustable autoloading.

I was trying to keep those errors in the same file for convenience, but failed to come up good solution that does not change namespaces.

In the future, when you can afford breaking API, suggestion would be dropping `errors` folder collapsing and use `Errors` namespace: `MatrixSdk::MatrixConflictError` => `MatrixSdk::Errors::ConflictError`, but that just a random suggestion, not based on anything other than personal preferences.

Backstory to this, was that I discovered one test that was failing when called alone via `TEST=test/api_cs_protocol_verification_test` and instead of fixing existing manual autoloading code, I went ahead and pulled in Zeitwerk to do its werk.

### CODE REVIEW

File: matrix-sdk.gemspec [hunk 1]
```diff
@@ -20,6 +20,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rake'
   spec.add_development_dependency 'simplecov'
   spec.add_development_dependency 'test-unit'
+  spec.add_development_dependency 'zeitwerk'
```
Comment (ilvez) [inline matrix-sdk.gemspec:23]: Oh, I guess this is normal dependency?

Comment (ilvez) [reply matrix-sdk.gemspec]: Fixed..

**Comment by ananace** [comment]
To be perfectly honest, the normal use of the SDK is going to load basically everything anyway.  
Just creating a `Client` instance is by design going to touch everything but the `AS`/`SS` protocols - and even then there's at least one `SS` protocol method I've been planning on dragging in there to get the server version.
The only reason autoloading was used in the first place is just because that's what I normally do when I write library gems.

So I think a more appropriate solution would just be to drop autoloading entirely, require everything out of the gate, and skip pulling in an additional dependency to solve a non-existent problem.

**Comment by ilvez** [comment]
Dropping autoloading and using regular requiring is optimal for a project of this size indeed.

**Pull request closed without merging**